pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -180,12 +191,39 @@ func (c *K8sOrchestrator) InitTopologyServiceInController(ctx context.Context) (
 					return nil, err
 				}
 
+				isAcceptPreferredDatatsoresFSSEnabled := c.IsFSSEnabled(ctx, common.AcceptPreferredDatastores)
+
 				controllerVolumeTopologyInstance = &controllerVolumeTopology{
-					k8sConfig:                 config,
-					nodeMgr:                   nodeManager,
-					csiNodeTopologyInformer:   *crInformer,
-					clusterFlavor:             clusterFlavor,
-					isCSINodeIdFeatureEnabled: c.IsFSSEnabled(ctx, common.UseCSINodeId),
+					k8sConfig:                             config,
+					nodeMgr:                               nodeManager,
+					csiNodeTopologyInformer:               *crInformer,
+					clusterFlavor:                         clusterFlavor,
+					isCSINodeIdFeatureEnabled:             c.IsFSSEnabled(ctx, common.UseCSINodeId),
+					isAcceptPreferredDatastoresFSSEnabled: isAcceptPreferredDatatsoresFSSEnabled,
+				}
+
+				if isAcceptPreferredDatatsoresFSSEnabled {
+					// Get CNS config.
+					cnsCfg, err := common.GetConfig(ctx)
```

Comment:
In the follow-up PR can you honor the change made in the vSphere config secret for `CSIFetchPreferredDatastoresIntervalInMin` and appropriately re-initialize the ticker.